src/workerd/util/duration-alert.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "kj/string.h"
+#include "kj/time.h"
+#include <kj/debug.h>
+
+class durationAlert {
+// This is a utility class for instantiating a timer, which will log if it's destructed after a specified time
+// This works by relying on RAII (Scope-Bound Resource Management) to check how much time has elapsed
+// when the object is destructed. This ensures that the time is checked when the timer object goes out of scope
+// thereby timing everything after the initialization of the timer, until the end of the scope where it was
+// instantiated.
+public:
+  durationAlert(const kj::MonotonicClock& clock, kj::Duration logWarningTime, kj::StringPtr logMessage)
+    : logWarningTime(logWarningTime),
+      logMessage(logMessage),
+      start(clock.now()),
+      clock(clock) {}
+
+  ~durationAlert() {
+    if (start - clock.now() >= logWarningTime) {
```

Comment:
Might have meant `clock.now() - start` ?

Reply:
If you wanted to write a `KJ_TEST` unit test to confirm this class works as expected, you might be able to use the same mechanism as the kj debug test to capture the log output (declare an instance of `ExceptionCallback` on the stack implementing `logMessage()`):

https://github.com/capnproto/capnproto/blob/958ccfe/c%2B%2B/src/kj/debug-test.c%2B%2B#L168-L175

...along with using `kj::TimerImpl::advanceTo()` to simulate advancing the clock without actually waiting, e.g.:

https://github.com/capnproto/capnproto/blob/958ccfe4/c%2B%2B/src/kj/compat/http-test.c%2B%2B#L3843

Reply:
I added a unit test based on KJ_EXPECT_LOG :slightly_smiling_face: .

I wanted to test the negative case as well (no log emitted when the duration was not exceeded), but that'll require some KJ changes, and doesn't seem quite as critical to have a test for. I might add that after making a change to KJ

Reply:
The test looks good, thanks!